packages/stokhos/src/sacado/kokkos/vector/Stokhos_MP_Vector_MaskTraits.hpp

```diff
@@ -0,0 +1,447 @@
+// @HEADER
+// ***********************************************************************
+//
+//                           Stokhos Package
+//                 Copyright (2009) Sandia Corporation
+//
+// Under terms of Contract DE-AC04-94AL85000, there is a non-exclusive
+// license for use of this work by or on behalf of the U.S. Government.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY SANDIA CORPORATION "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Eric T. Phipps (etphipp@sandia.gov).
+//
+// ***********************************************************************
+// @HEADER
+
+#ifndef STOKHOS_MP_VECTOR_MASKTRAITS_HPP
+#define STOKHOS_MP_VECTOR_MASKTRAITS_HPP
+
+#include "Sacado.hpp"
+#include "Stokhos_Tpetra_Utilities_MP_Vector.hpp"
```

Comment:
This code in now way depends on Tpetra, so different includes should be used.  I think just including Sacado_MP_Vector.hpp should be sufficient.

Reply:
Yes, I do agree, however, replacing 
```
#include "Stokhos_Tpetra_Utilities_MP_Vector.hpp"
```
by
```
#include "Sacado_MP_Vector.hpp"
```
leads to the building error included in this log file:

[log_Mask.txt](https://github.com/trilinos/Trilinos/files/1968095/log_Mask.txt)

Reply:
There are two ways to work around: 
1) Include Stokhos_Sacado_Kokkos_MP_Vector.hpp
2) Include Sacado_MP_Vector.hpp but remove the value_type from struct EnsembleTraits_m and replace every ET::coeff(v,i) by v[i].

Either way is fine for me, let me know which one you prefer.

Reply:
Including Stokhos_Sacado_Kokkos_MP_Vector.hpp should be fine.